sycl/source/detail/program_manager/program_manager.cpp

```diff
@@ -1866,34 +1866,44 @@ void ProgramManager::addImages(sycl_device_binaries DeviceBinary) {
 
     m_BinImg2KernelIDs[Img.get()].reset(new std::vector<kernel_id>);
 
-    for (sycl_offload_entry EntriesIt = EntriesB; EntriesIt != EntriesE;
-         ++EntriesIt) {
+    sycl_offload_entry EntriesIt;
+    auto IncrementEntriesIt = [&]() {
+      if (EntriesIt->IsNewOffloadEntryType())
```

Comment:
We shouldn't expect a mix of old and new entries, so I think this check should be made once instead of on each increment.

Reply:
Done in https://github.com/intel/llvm/pull/17109/commits/8d706016b4e55d792453dff255b23ee05c883348